File: Rio/Rio.Web/Modules/ServerTypes/Workspace/GroupForm.ts
import { StringEditor, LookupEditor, TextAreaEditor, PrefixedContext } from "@serenity-is/corelib";
import { initFormType } from "@serenity-is/corelib/q";

export interface GroupForm {
    Name: StringEditor;
    ParentId: LookupEditor;
    Description: TextAreaEditor;
    SelectedTenant: LookupEditor;
    TeacherId: LookupEditor;
}

export class GroupForm extends PrefixedContext {
    static formKey = 'Workspace.Group';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!GroupForm.init)  {
            GroupForm.init = true;

            var w0 = StringEditor;
            var w1 = LookupEditor;
            var w2 = TextAreaEditor;

            initFormType(GroupForm, [
                'Name', w0,
                'ParentId', w1,
                'Description', w2,
                'SelectedTenant', w1,
                'TeacherId', w1
            ]);
        }
    }
}

File: Rio/Rio.Web/Modules/Workspace/Group/GroupForm.cs
using Serenity.ComponentModel;
using Serenity.Web;
using System;

namespace Rio.Workspace.Forms
{
    [FormScript("Workspace.Group")]
    [BasedOnRow(typeof(GroupRow), CheckNames = true)]
    public class GroupForm
    {
        [HalfWidth]
        public string Name { get; set; }
        [HalfWidth]
        public int ParentId { get; set; }
        [TextAreaEditor(Rows = 3)]
        public string Description { get; set; }
        [HalfWidth]
        public int SelectedTenant { get; set; }
        [HalfWidth]
        public int TeacherId { get; set; }
    }
}